c10/util/BFloat16.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+// Defines the bloat16 type (brain floating-point). This representation uses
+// 1 bit for the sign, 8 bits for the exponent and 7 bits for the mantissa.
+
+#include <c10/macros/Macros.h>
+#include <cmath>
+#include <cstring>
+
+namespace c10 {
+
+namespace detail {
+  inline C10_HOST_DEVICE float f32_from_bits(uint16_t src) {
+    float res = 0;
+    uint32_t tmp = src;
+    tmp <<= 16;
+
+#ifdef __HIP_PLATFORM_HCC__
+    float* tempRes = 0;
```

Comment:
This is assigning 0 to a pointer type, which I guess is fine but looks kinda weird